compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/BIRGen.java

```diff
@@ -2442,6 +2403,146 @@ public void visit(BLangRegExpTemplateLiteral regExpTemplateLiteral) {
         this.env.targetOperand = toVarRef;
     }
 
+    private BIRNonTerminator.NewStructure createNewStructureInst(BIROperand typeDesc,
+                                                                 List<BIRNode.BIRMappingConstructorEntry> fields,
+                                                                 BIROperand toVarRef, Location pos) {
+        return new BIRNonTerminator.NewStructure(pos, toVarRef, typeDesc, fields);
+    }
+
+    public String getTypedescFieldName(String name) {
+        return "$typedesc$" + name;
+    }
+
+    private BIRVariableDcl getTypedescVariable(BType type, Location pos) {
+        Supplier<BIRVariableDcl>[] checks = new Supplier[] {
```

Comment:
Do we really need this? Given that we don't anticipate additions, can just call and see in order?

Reply:
Yes that was the previous code. But it was lengthy. A con I feel with this approach is its hard to read and debug. I'm +1 to go with any approach. WDYT?

Reply:
This is changed to simplified version